src/Field.php

```diff
@@ -489,7 +489,12 @@ public function setNull(): self
      */
     public function compare($value): bool
     {
-        return $this->get() == $value;
+        if ($this->owner->persistence === null) {
```

Comment:
what do to with this if? It will fail if normalize returns an object which may not have `__toString()` implemented (like `\DateTime`)